src/vm.rs

```diff
@@ -131,7 +131,7 @@ impl InstructionMeter for DefaultInstructionMeter {
 /// ```
 pub struct EbpfVm<'a, E: UserDefinedError> {
     executable: &'a dyn Executable<E>,
-    compiled_prog: Option<JitProgram<E>>,
+    compiled_prog_and_arg: Option<(JitProgram<E>, Vec<*const u8>)>,
```

Comment:
why deliminate between `compiled_prog` and `compiled_prog_and_arg`, can we just switch the definition of `compiled_prog` to contain args?

Reply:
Unfortunately no, because the argument is VM specific as it contains the memory mapping.